text/0013-delete-enqueued-updates.md

```diff
@@ -0,0 +1,128 @@
+- Start Date:
+- Specification PR: https://github.com/meilisearch/specifications/pull/13
+- MeiliSearch Issue: 
+    - https://github.com/meilisearch/MeiliSearch/issues/1104
+- Parent Spec: https://github.com/meilisearch/specifications/pull/12
+
+
+# Abort enqueued updates
+
+## 1. Feature Description and Interaction
+
+### Summary
+
+Allow to abort messages in the update queue or to empty the entire queue. A abortd update will have the status `aborted`.
+
+Questions not answered:
+- We have an attribute on update info `processedAt`. What becomes this attribute if we add the possibility to abort an update? We already use this term in the case of a failed update.
+
+### Motivation
+
+The update queue is convenient because it allows us to serialize updates and assure the [ACID property](https://en.wikipedia.org/wiki/ACID) of MeiliSearch. It also causes some problems. As the queue message is an append-only queue, it is impossible to remove an update from this queue. The updates can be large and can take a long time to index. Sometimes, it would be convenient to abort a message in this queue or empty it. 
+
+### Additional Materials
+
+As explained on the [parent spec](https://github.com/meilisearch/specifications/pull/12), It's quite rare to have this kind of update queue. Elasticsearch or Algolia, who have this king of update queue, don't give the abort or clear their 'tasks' queues. 
+
+### Explanation
+
+To make this feature possible, two new routes and a new status (`aborted`) must be added. 
+
+An aborted update will stay at its place on the queue. So the typical answer of the get all update status route will be: 
+
+```json
+[
+    {
+        "status": "processed",
+        "updateId": 1,
+        ...
+    },
+    {
+        "status": "failed",
+        "updateId": 2,
+        ...
+    },
+    {
+        "status": "processed",
+        "updateId": 3,
+        ...
+    },
+    {
+        "status": "enqueued",
+        "updateId": 4,
+        ...
+    },
+    {
+        "status": "aborted",
+        "updateId": 5,
+        ...
+    },
+    {
+        "status": "enqueued",
+        "updateId": 6,
+        ...
+    },
+    ...
+]
+```
+
+the update status document will look like:
+
+```json
+{
+  "status": "aborted",
+  "updateId": 12,
+  "type": {
+    "name": "DocumentsAddition",
+    "number": 4
+  },
+  "duration": 0,
+  "enqueuedAt": "2019-12-07T21:16:09.623944Z",
+  "processedAt": "2019-12-07T21:16:09.703509Z" // Date of deletion
+}
+```
+
+#### HTTP API
+
+**Abort an update:**
+
+- Method: DELETE
+- Route: `/indexes/:index_uid/updates/:update_id`
+    - index_uid: The index uid
+    - update_id: The update identifier
+- Status Code: 204
+- Response: No Body
+
+**Abort all update:**
+
+- Method: DELETE
+- Route: `/indexes/:index_uid/updates`
+    - index_uid: The index uid
+- Status Code: 204
+- Response: No Response
+
+#### Potential errors
+
+- The given id is a processed id: `400` - `A processed/failed update cannot be abortd`
```

Comment:
not clear, what should the response look like on failed deletion attempt? What should the meili error code be? What http error code?

More generally what type of error can we encounter?

Reply:
Yes agree, I'll work on that!

Reply:
Updated! Don't hesitate to give feedback on error messages and directly propose changes.